File: src/main/java/voidchess/player/PlayerInterface.java
package voidchess.player;

import voidchess.helper.PawnPromotion;
import voidchess.helper.Position;

/**
 * @author stephan
 */
public interface PlayerInterface {
    void setIsPlaying(boolean isPlaying);

    void play();

    PawnPromotion askForPawnPromotionType(Position pawnPosition);

    void setColor(boolean isWhite);
}

File: src/main/kotlin/voidchess/player/PlayerInterface.kt
package voidchess.player

import voidchess.board.MoveResult
import voidchess.helper.PawnPromotion
import voidchess.helper.Position


interface PlayerInterface {
    fun gameStarts()
    fun gameEnds(endoption: MoveResult, lastMoveByWhite: Boolean)
    fun gaveCheck()
    fun play()
    fun askForPawnPromotionType(pawnPosition: Position): PawnPromotion
    fun setColor(isWhite: Boolean)
}
